qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -100,16 +103,30 @@ def __init__(
         self._circuit.compose(self.feature_map, inplace=True)
         self._circuit.compose(self.ansatz, inplace=True)
 
-        # construct circuit QNN
-        neural_network = CircuitQNN(
-            self._circuit,
-            input_params=self.feature_map.parameters,
-            weight_params=self.ansatz.parameters,
-            interpret=self._get_interpret(2),
-            output_shape=2,
-            quantum_instance=quantum_instance,
-            input_gradients=False,
-        )
+        # needed for mypy
+        neural_network: SamplerQNN | CircuitQNN = None
+        if sampler is None:
+            # construct circuit QNN
+            neural_network = CircuitQNN(
```

Comment:
I'd construct `SamplerQNN` if sampler is none with the defaults. And only if `quantum_instance` is not none fallback to `CircuitQNN`. Basically, it should be preferable to create `SamplerQNN` unless we can't.